docs/site/BelongsTo-relation.md

```diff
@@ -75,6 +75,14 @@ export interface OrderRelations {
 export type OrderWithRelations = Order & OrderRelations;
 ```
 
+
+Definition of `belongsTo` relation can also include explicit `keyFrom`, `keyTo` and `name` attributes.
+
+- keyFrom is foreign key of the referenced model, whereas keyTo is primary key of referenced model, name is the referenced model name. Usage of keyFrom and keyTo in a belongsTo relation is similar to the use in legacy datasource juggler.
+- In the above example for the model Order, keyFrom is `customerId` and keyTo is `id`. keyFrom defaults to `{decorated-belongsTo-property-name}`, keyTo defaults to `id` and the referenced model name defaults to the decorated property name after stripping the trailing `Id` suffix.
+- In case in the above example, the property name `customerId` has to be customized to say `cust_id`, then the keyFrom,  keyTo and name values needs to be explicitly defined as `@belongsTo(() => Customer, { keyFrom: 'customerId', keyTo: 'id'}, { name: 'customer' })`.
```

Comment:
Please don't use inline code block for `@belongsTo` example, use triple-back-ticks and show the full property definition.

For example:

```ts
@belongsTo(
  () => Customer,
  {keyFrom: 'cust_id', keyTo: 'id'},
  {name: 'customer'}
)
cust_id: string;
```

Please note that you have a bug in your example - the `name` property belongs to the second decorator argument (relation metadata). The third argument is modifying property definition of the foreign key.

I believe the example above can be simplified as follows:

```ts
@belongsTo(
  () => Customer,
  {name: 'customer'}
)
cust_id: string;
```

Did you perhaps want to show how to use a foreign key property with a name different from the database column?

I believe that's defined as follows:

```ts
@belongsTo(
  () => Customer,
  {
    name: 'customer',  // relation name
    keyFrom: 'cust_id', // PK property name
  },
  {
    name: 'cust_id', // PK property name
  },
)
customerId: string;
```

Since we don't have any tests to verify that our recommended configuration actually works, I'd prefer to leave this part out of your pull request.

Reply:
@bajtos , @agnes512 , if required I can add test cases. But we cannot leave this part out of this PR. This issue originated for belongsTo documentation with all the above gaps.

Reply:
I am fine to leave the test out of this PR.
Please make sure the explanations are correct, and the given examples are clear.

Besides, the workaround given by the community:
```
 @belongsTo(
    () => UserTenant,
    {keyFrom: 'created_by', name: 'created_by'},
    {
      name: 'created_by',
    },
  )
  createdBy?: number;
```
might work. But I don't think it's a proper way to use `keyFrom` and `keyTo`. @bajtos 's second example should be enough for customizing our own foreign key names. I also have some explanations in my other comments.

You can also check files: `loopback-next/packages/repository/src/relations/belongs-to/belongs-to.decorator.ts` and `loopback-next/packages/repository/src/relations/has-many/has-many.decorator.ts` to verify the usage of keyfrom & keyTo.

Reply:
@agnes512 @bajtos , keyFrom has become a dummy parameter with the recent changes. I have simplified your comments as parameter definition in the latest commit.